File: code_challenge_solution/java/20 Calculate Dot and Cross Product of Two Vectors.java
import java.util.Arrays;

public class VectorMath {
    public static void main(String[] args) {
        // First vector
        int[] vector1 = {2, 3, 4};
        
        // Second vector
        int[] vector2 = {5, 6, 7};
        
        // Calculate dot product
        int dotProduct = calculateDotProduct(vector1, vector2);
        System.out.println("Dot Product: " + dotProduct);
        
        // Calculate cross product
        int[] crossProduct = calculateCrossProduct(vector1, vector2);
        System.out.println("Cross Product: " + Arrays.toString(crossProduct));
    }
    
    public static int calculateDotProduct(int[] vector1, int[] vector2) {
        if (vector1.length != vector2.length) {
            throw new IllegalArgumentException("Vectors must have the same size");
        }
        
        int dotProduct = 0;
        for (int i = 0; i < vector1.length; i++) {
            dotProduct += vector1[i] * vector2[i];
        }
        
        return dotProduct;
    }
    
    public static int[] calculateCrossProduct(int[] vector1, int[] vector2) {
        if (vector1.length != 3 || vector2.length != 3) {
            throw new IllegalArgumentException("Vectors must have size 3 for cross product calculation");
        }
        
        int x = vector1[1] * vector2[2] - vector1[2] * vector2[1];
        int y = vector1[2] * vector2[0] - vector1[0] * vector2[2];
        int z = vector1[0] * vector2[1] - vector1[1] * vector2[0];
        
        int[] crossProduct = {x, y, z};
        return crossProduct;
    }
}

File: code_challenge_solution/c#/20 Calculate Dot and Cross Product of Two Vectors.cs
```csharp

using System;

namespace VectorMath
{
    class Program
    {
        static void Main(string[] args)
        {
            int[] vector1 = { 1, 2, 3 }; // vector 1
            int[] vector2 = { 4, 5, 6 }; // vector 2

            // Calculate dot product of the two vectors
            int dotProduct = CalculateDotProduct(vector1, vector2);
            Console.WriteLine("Dot product: " + dotProduct);

            // Calculate cross product of the two vectors
            int[] crossProduct = CalculateCrossProduct(vector1, vector2);
            Console.WriteLine("Cross product: (" + crossProduct[0] + ", " + crossProduct[1] + ", " + crossProduct[2] + ")");
        }

        static int CalculateDotProduct(int[] vector1, int[] vector2)
        {
            int dotProduct = 0;

            for (int i = 0; i < vector1.Length; i++)
            {
                dotProduct += vector1[i] * vector2[i];
            }

            return dotProduct;
        }

        static int[] CalculateCrossProduct(int[] vector1, int[] vector2)
        {
            int[] crossProduct = new int[3];

            crossProduct[0] = vector1[1] * vector2[2] - vector1[2] * vector2[1];
            crossProduct[1] = vector1[2] * vector2[0] - vector1[0] * vector2[2];
            crossProduct[2] = vector1[0] * vector2[1] - vector1[1] * vector2[0];

            return crossProduct;
        }
    }
}

```